### DISCUSSION TITLE
Discussion about Milestones

### DISCUSSION THREAD

**Discussion by kkuepper** [bcc-code/bmm-web#156] [General]
Here we can discuss the milestones https://github.com/bcc-code/bmm-web/issues/154.
@SimonSimCity @sifferhans

**Comment by kkuepper** [comment]
I moved the player to v0.2 since I overlooked the next/previous buttons.
By player I mean
- design
- next / previous
- jump 15 seconds
- show current track & cover
- repeat
- simple version of shuffle

Not part of player:
- smarter shuffle
- Queue editing
- adding to queue

**Reply by SimonSimCity** [reply]
I've spoken about this with Leo and we agreed to have nothing as fixed, but allow a rewrite of the player as it is now. For much of the code added in https://github.com/bcc-code/bmm-web/pull/104 I'm not quite happy with because there are too many things not taken into account.

Now to the things you wrote ... For next/previous, repeat and shuffle to work, we need a basic implementation of a queue. Last time we spoke, we agreed on to take a very basic queuing system, which is in line with what I found at YT-Music, so we'll align with their system.

For "not part of player" - what do you think of when speaking about smarter shuffle ..? Do you mean https://medium.com/immensity/how-spotifys-shuffle-algorithm-works-19e963e75171?

**Reply by kkuepper** [reply]
by smart shuffle, I mean that the original queue is restored if you unshuffle. As far as I'm concerned we already have a basic queue implementation.
